### PR TITLE
FST-based labels (initial class)

### PR DESCRIPTION
**Pull request by tarekziade**
### Description

Introduces a new `nucliadb_vectors::labels::LabelDictionnary` class that can be used to index and retrieve labels and for each a list of document ids

### How was this PR tested?

- unit tests, 100% covered
- benchmark in `vectors_benchmark`

### CODE REVIEW

**Comment by tarekziade** [comment]
Run in the CI for 100k labels:

- index time: ~2.5 seconds
- search time : 400 microseconds

Screenshot 2023-08-17 at 16 12 59

File: nucliadb_vectors/src/labels/mod.rs [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    for label in labels {
+        let start_position: u64 = records_writer.stream_position()?;
+        serialize_into(&mut records_writer, label)?;
+        fst_builder.insert(label.key.clone(), start_position)?;
+        added += 1;
+    }
+    let _ = records_writer.flush();
+    fst_builder.finish()?;
+    Ok(added)
+}
```
Comment (hermeGarcia) [inline nucliadb_vectors/src/labels/mod.rs:139]: I tend to code this as type functions, to not lose the connection between the files and the Dictionary:
```rust
impl LabelDictionary {
    pub fn new<'a, I>(fst_file_path: &str, records_file_path: &str, labels: I) -> VectorR<Self>
    where
        I: Iterator<Item = &'a Label>,
    {
           // build_index logic but returning the new LabelDictionary
    }
    pub fn open(fst_file_path: &str, records_file_path: &str) -> VectorR<Self> {
        let records_file = File::open(records_file_path)?;
        let mmap = unsafe { Mmap::map(&File::open(fst_file_path)?)? };
        let fst_map = Map::new(mmap)?;

        Ok(Self {
            fst_file_path: fst_file_path.to_owned(),
            records_file_path: records_file_path.to_owned(),
            records_file: RwLock::new(records_file),
            fst_map,
        })
    }
}
```

File: nucliadb_vectors/src/labels/mod.rs [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+    // 2. add the key into the fst file, the value is the offset in the record file
+    let mut added = 0;
+
+    for label in labels {
+        let start_position: u64 = records_writer.stream_position()?;
+        serialize_into(&mut records_writer, label)?;
+        fst_builder.insert(label.key.clone(), start_position)?;
+        added += 1;
+    }
+    let _ = records_writer.flush();
```
Comment (hermeGarcia) [inline nucliadb_vectors/src/labels/mod.rs:136]: why swallow the error instead of `records_writer.flush()?;` ?

**Review by hermeGarcia — APPROVED** [review]
LGTM just some comments

**Pull request merged**